### DISCUSSION TITLE
Colang v2 more examples

### DISCUSSION THREAD

**Discussion by alessandro-neri** [NVIDIA-NeMo/Guardrails#783] [Q&A]
Hello, I am having trouble implementing colang v2 beyond the very simple examples provided in the documentation, especially regarding how to trigger actions. Additionally, I can't find much information regarding the input rails. Can we still set up input rails in the yaml config file, or is there now only one way to do it? Could anyone point me to more examples or help me figure this out?

This is the only example I found regarding input rails.
And also regarding actions I haven't found much.

```flow input rails $input_text
  $input_safe = await check user utterance $input_text

  if not $input_safe
    bot say "I'm sorry, I can't respond to that."
    abort

flow check user utterance $input_text -> $input_safe
  $is_safe = ..."Consider the following user utterance: '{$input_text}'. Assign 'True' if appropriate, 'False' if inappropriate."
  print $is_safe
  return $is_safe```

**Comment by Pouyanpi** [comment]
Hi @alessandro-neri 

Please have a look at this [example](https://github.com/NVIDIA/NeMo-Guardrails/tree/develop/examples/bots/abc_v2). So this is abc bot in Colang 2.x where you can see how input rails and output rails are used. In 0.10.0 we have the support for NeMo Guardrails standard library including self check input etc. I hope it helps.

**Reply by alessandro-neri** [reply]
Thank you, the examples are still quite basic... do you have any examples of more complex flows? And especially executing actions. @Pouyanpi 

I am trying to use it in a company project, but I will need to use other frameworks otherwise.
